python/erdos/streams.py

```diff
@@ -58,6 +59,147 @@ def name(self) -> str:
+            function.
+        """
+
+        def flat_map_fn(serialized_data: bytes) -> Sequence[bytes]:
+            mapped_values = function(pickle.loads(serialized_data))
+            result = []
+            for element in mapped_values:
+                result.append(pickle.dumps(element))
+            return result
+
+        return OperatorStream(self._internal_stream._flat_map(flat_map_fn))
+
+    def filter(self, function: Callable[[Any], bool]) -> "OperatorStream":
+        """Applies the given function to each received value on the stream, and outputs
+        the value if the function evaluates to true.
+
+        Args:
+            function (Callable[[Any], bool]): The function to be applied to each
+                message received on the input stream.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the filtered results from the
+            applied function.
+        """
+
+        def filter_fn(serialized_data: bytes) -> bool:
+            return function(pickle.loads(serialized_data))
+
+        return OperatorStream(self._internal_stream._filter(filter_fn))
+
+    def split(
+        self, function: Callable[[Any], bool]
+    ) -> Tuple["OperatorStream", "OperatorStream"]:
+        """Applies the given function to each received value on the stream, and outputs
+        the value to either the left or the right stream depending on the returned
+        boolean value from the function.
+
+        Args:
+            function (Callable[[Any], bool]): The function to be applied to each
+                message received on the input stream.
+
+        Returns:
+            A Tuple[:py:class:`OperatorStream`, :py:class:`OperatorStream`] that carry
+            the results output to the left and right streams respectively.
+        """
+
+        def split_fn(serialized_data: bytes) -> bool:
+            return function(pickle.loads(serialized_data))
+
+        left_stream, right_stream = self._internal_stream._split(split_fn)
+        return (OperatorStream(left_stream), OperatorStream(right_stream))
+
+    def timestamp_join(self, other: "Stream") -> "OperatorStream":
+        """Joins the data with matching timestamps from the two different streams.
+
+        Args:
+            other (:py:class:`Stream`): The other stream that needs to be joined with
+                self.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the joined results from the two
+            streams.
+        """
+
+        def join_fn(serialized_data_left: bytes, serialized_data_right: bytes) -> bytes:
+            left_data = pickle.loads(serialized_data_left)
+            right_data = pickle.loads(serialized_data_right)
+            return pickle.dumps((left_data, right_data))
+
+        return OperatorStream(
+            self._internal_stream._timestamp_join(other._internal_stream, join_fn)
+        )
+
+    def concat(self, other: Union["Stream", Sequence["Stream"]]) -> "OperatorStream":
```

Comment:
```suggestion
    def concat(self, *other: "Stream") -> "OperatorStream":
```
IMO this is a good application of variadics.